notebooks/Tutorial9-Multitable.py

```diff
@@ -0,0 +1,521 @@
+    "Column Direct Match Protection": [
+        round(d.column_direct_match_protection, 2) for d in data
+    ],
+    "Categorical Hidden Rate": [round(d.categorical_hidden_rate, 2) for d in data],
+    "Row Direct Match Protection": [
+        round(d.row_direct_match_protection, 2) for d in data
+    ],
+    "Table": [
+        d.dataset_name
+        + "_"
+        + d.computation_type.name
+        + (
+            "_" + d.individual_identifier_variable
+            if d.computation_type.name != "standalone"
+            else ""
+        )
+        for d in data
+    ],
+}
+
+summary_table = pd.DataFrame(data_dict)
+summary_table.set_index("Table", inplace=True)
+summary_table = df.reindex(index)
+summary_table.loc["TARGET", :] = [90, 5, 0.2, 0.3, 50, 90, 90]
+# -
+
+df
+
+# # Utility evaluation
+
+# ## Univariate comparison
+#
+# You can here compare distributions between original and avatar data.
+# If distributions are not well preserved, you can work with the parameter `columns_weight` in your avatarization.
+
+# ### Patient
+
+# +
+patient_combined = pd.concat([patient, patient_avatar]).reset_index(drop=True)
+patient_combined["type"] = np.repeat(["original", "avatar"], 150)
+map_color = {"original": "dimgrey", "avatar": "#3BD6B0"}
+# Distribution continuous sur la même sortie
+fig, axes = plt.subplots(2, 2, figsize=(12, 8))
+sns.histplot(
+    data=patient_combined,
+    x="height",
+    hue="type",
+    palette=map_color,
+    kde=True,
+    ax=axes[0, 0],
+    stat="density",
+)
+sns.histplot(
+    data=patient_combined,
+    x="weight",
+    hue="type",
+    palette=map_color,
+    kde=True,
+    ax=axes[0, 1],
+    stat="density",
+)
+sns.histplot(
+    data=patient_combined,
+    x="age",
+    hue="type",
+    palette=map_color,
+    kde=True,
+    ax=axes[1, 0],
+    stat="density",
+)
+sns.countplot(
+    data=patient_combined, x="gender", hue="type", palette=map_color, ax=axes[1, 1]
+)
+
+axes[0, 0].set_title("Height Distribution")
+axes[0, 1].set_title("Weight Distribution")
+axes[1, 0].set_title("Age Distribution")
+axes[1, 1].set_title("Gender Distribution")
+plt.tight_layout()
+plt.show()
+# -
+
+# ### Doctor
+
+# +
+doctor_combined = pd.concat([doctor, doctor_avatar]).reset_index(drop=True)
+doctor_combined["type"] = np.repeat(["original", "avatar"], 50)
+map_color = {"original": "dimgrey", "avatar": "#3BD6B0"}
+# Distribution continuous sur la même sortie
+fig, axes = plt.subplots(1, 2, figsize=(12, 4))
+sns.histplot(
+    data=doctor_combined,
+    x="age",
+    hue="type",
+    palette=map_color,
+    kde=True,
+    ax=axes[0],
+    stat="density",
+)
+sns.countplot(data=doctor_combined, x="job", hue="type", palette=map_color, ax=axes[1])
+
+axes[0].set_title("Age Distribution")
+axes[1].set_title("Job Distribution")
+plt.tight_layout()
+plt.show()
+# -
+
+# ### Visit
+
+visit_combined = pd.concat([visit, visit_avatar]).reset_index(drop=True)
+visit_combined["type"] = np.repeat(["original", "avatar"], 300)
```

Comment:
```suggestion
visit["type"] = "original"
visit_avatar["type"] = "avatar"
visit_combined = pd.concat([visit, visit_avatar]).reset_index(drop=True)
```
I feel stuff like this is more secure, what do you think? (I am always afraid to mix up avatar and original)

Reply:
Don't worry both are working the same. No risk of mixing here. Concat is basically putting a dataset on top of another and the documentation of np.repeat available here (https://numpy.org/doc/stable/reference/generated/numpy.repeat.html) makes sure that we have all original then all avatars.